apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,380 @@
+        /// The maximum number of cells allowed per row.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter CellsPerRowLimit(int limit) =>
+            new RowFilter { CellsPerRowLimitFilter = limit };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which skips the first N cells
+        /// of each row, matching all subsequent cells.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If duplicate cells are present, as is possible when using an Interleave,
+        /// each copy of the cell is counted separately.
+        /// </para>
+        /// </remarks>
+        /// <param name="offset">
+        /// The number of cells to skip per row.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter CellsPerRowOffset(int offset) =>
+            new RowFilter { CellsPerRowOffsetFilter = offset };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which sends rows through
+        /// several RowFilters in sequence.
+        /// </summary>
+        /// <param name="filters">
+        /// The elements of "filters" are chained together to process the input row:
+        /// in row -> f(0) -> intermediate row -> f(1) -> ... -> f(N) -> out row
+        /// The full chain is executed atomically.  Must not be null, or contain null
+        /// elements.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Chain(params RowFilter[] filters) =>
+            new RowFilter
+            {
+                Chain = new RowFilter.Types.Chain
+                {
+                    Filters = { Utilities.ValidateCollection(filters, nameof(filters)) }
+                }
+            };
+        
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells from
+        /// columns whose qualifiers satisfy the given RE2 regex.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that, since column qualifiers can contain arbitrary bytes, the `\C`
+        /// escape sequence must be used if a true wildcard is desired. The `.`
+        /// character will not match the new line character `\n`, which may be
+        /// present in a binary qualifier.
+        /// </para>
+        /// <para>
+        /// Note that string is implicitly convertible to <see cref="BigtableByteString"/>,
+        /// so <paramref name="regex"/> can be specified using a string as well and its UTF-8
+        /// representations will be used.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column qualifiers.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnQualifierRegex(BigtableByteString regex) =>
```

Comment:
Out of interest, what's the benefit in this accepting `BigtableByteString` rather than just string? When do we expect anyone to pass anything *other* than a `string` in? My guess is that using a value from one call as a filter input in another, but I thought I'd check..

Reply:
This is to match the API specification (see column_qualifier_regex_filter [here](https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#google.bigtable.v2.RowFilter)). I assumed you could just specify an arbitrary byte string to find values that match that, but I guess it would always try to interpret it at RE2, so perhaps string is best.